### PR TITLE
[build] Docker build environment for building VM

### PR DESCRIPTION
**Pull request by DmytroRozzuvailo**
Signed-off-by: Dmytro Rozzuvailo <dmytro_rozzuvailo@epam.com>

reviewers: @al1img @MykolaSolyanko @Eabramov84

### CODE REVIEW

File: qemu_start.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# !/bin/sh
```
Comment (al1img) [inline qemu_start.sh:1]: This script exists in meta-aos-vm/doc

File: run_docker.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        ;;
+    # -- means the end of the arguments; drop this, and break out of the while loop
+    --)
+        shift
+        break
+        ;;
+    # If invalid options were passed, then getopt should have reported an error,
+    # which we checked as VALID_ARGUMENTS when getopt was called...
+    *)
+        echo "Unexpected option: $1 - this should not happen."
```
Comment (al1img) [inline run_docker.sh:51]: But it happens, so display the proper error here.

File: run_docker.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        usage
+        ;;
+    esac
+done
+
+# Check we have Docker image and workspace or we can't work.
+if [ -z "${DOCKER_IMAGE_NAME}" ] || [ -z "${WORKSPACE}" ]; then
+    usage; exit 100
+else
```
Comment (al1img) [inline run_docker.sh:60]: Why do you need else after exit?
Why exit 100?

File: run_docker.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+# Here you can change TAG for Docker image 
+IMAGE_TAG="latest"
+
+function usage() {
+    echo "Run Docker Container for building Yocto"
+    echo ""
+    echo "Command line arguments:"
+    echo -e "\t--workspace, -w\n\t\tWorkspace"
+    echo -e "\t--suffix, -s\n\t\tSuffix for Container name (optional)"
```
Comment (al1img) [inline run_docker.sh:13]: What is suffix here?

File: run_docker.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    echo -e "\t--docker_image, -d\n\t\tDocker image that should be used for building Yocto"
+    echo ""
+
+    exit 2
```
Comment (al1img) [inline run_docker.sh:17]: Why exit 2?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+# Creating a user
+RUN groupadd --gid "${USER_GID}" "${USER_NAME}" && \
+    useradd \
+      --uid ${USER_ID} \
+      --gid ${USER_GID} \
+      --create-home \
+      --shell /bin/bash \
+      ${USER_NAME}
+
+# Swit the user from root to $USER_NAME (yocto)
```
[the remark this answers is not in the document]
Comment (DmytroRozzuvailo) [reply Dockerfile]: fixed

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+# Define name for user and uid/gid
+ENV USER_NAME=yocto
+ARG USER_ID=1000
+ARG USER_GID=1000
+
+# Creating a user
```
[the remark this answers is not in the document]
Comment (DmytroRozzuvailo) [reply Dockerfile]: changed to 'the'

File: qemu_start.sh (hunk 1 above)
Comment (DmytroRozzuvailo) [reply qemu_start.sh]: removed

File: run_docker.sh (hunk 2 above)
Comment (DmytroRozzuvailo) [reply run_docker.sh]: after this message `usage` shows what exactly args should be used.
to do not make it unclear - removed part of message ` - this should not happen.`

File: run_docker.sh (hunk 3 above)
Comment (DmytroRozzuvailo) [reply run_docker.sh]: changed to `exit 1`

File: run_docker.sh (hunk 4 above)
Comment (DmytroRozzuvailo) [reply run_docker.sh]: you can add suffix to container name, but I guess it is not necessary now - removed this option

File: doc/run_docker.sh [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        ;;
+    esac
+done
+
+# Check we have Docker image and workspace or we can't work.
+if [ -z "${DOCKER_IMAGE_NAME}" ] || [ -z "${WORKSPACE}" ]; then
+    usage; exit 1
+fi
+
+if [ -z "${IMAGE_TAG}" ]; then
```
Comment (al1img) [inline doc/run_docker.sh:57]: IMAGE_TAG is set once and never changed. Why do you need to check it?

Comment (DmytroRozzuvailo) [reply doc/run_docker.sh]: forgot to remove that. done

File: README.md [hunk 9]
```diff
@@ -2,10 +2,33 @@
...
+## Prepare Docker Environment
+
+```sh
+cd doc
+
+# build docker image
+docker build . -f Dockerfile --build-arg "USER_ID=$(id -u)" --build-arg "USER_GID=$(id -g)" -t aos_yocto_image:latest
+
+# create directory for building VM
+mkdir -p ~/aos/build_aos_zephyr
```
Comment (al1img) [inline README.md:14]: change zephyr to vm for example

File: doc/run_docker.sh [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    *)
+        echo "Unexpected option: $1"
+        usage
+        ;;
+    esac
+done
+
+# Check we have Docker image and workspace or we can't work.
+if [ -z "${DOCKER_IMAGE_NAME}" ] || [ -z "${WORKSPACE}" ]; then
+    usage; exit 1
```
Comment (al1img) [inline doc/run_docker.sh:51]: Usage contains exit 1.

**Pull request merged**